### PR TITLE
TASK: De-Duplicate site and domain logic

### PR DESCRIPTION
**Pull request by kitsunet**
The logic to fetch site and domain for the ``ContentContext`` was
duplicated in several parts of Neos. This duplication is rather useless
as the values derived by it can be seen as default values for the two
properties. Therefore the logic was refactored to the
``ContentContextFactory`` and removed from other places in favor of
letting the factory set it as default.

### CODE REVIEW

**Comment by kitsunet** [comment]
@aertmann @kdambekalns on popular request ;)

**Comment by kdambekalns** [comment]
Looks good, but the failures need to be taken care of (they are real, not caused by some obscure PhpUnit behaviour…)

**Comment by kitsunet** [comment]
I know, I am on it, I have to move some mock objects around, so far I think it doesn't really show a failure of the code.

File: TYPO3.Neos/Tests/Unit/Domain/Service/ContentContextFactoryTest.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+class ContentContextFactoryTest extends \TYPO3\Flow\Tests\UnitTestCase
+{
+    /**
+     * @test
+     */
+    public function createWillSetDomainAndSiteFromCurrentRequestIfNotGiven()
+    {
+        $mockDomainRepository = $this->getMockBuilder(DomainRepository::class)->disableOriginalConstructor()->getMock();
+
+        $mockSite = $this->getMock(Site::class, [], [], '', false);
```
Comment (kdambekalns) [inline TYPO3.Neos/Tests/Unit/Domain/Service/ContentContextFactoryTest.php]: Please use `getMockBuilder(Site::class)->disableOriginalConstructor()->getMock()`

File: TYPO3.Neos/Tests/Unit/Domain/Service/ContentContextFactoryTest.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+        $mockDomain = $this->getMockBuilder(Domain::class)->disableOriginalConstructor()->getMock();
+        $mockDomain->expects($this->atLeastOnce())->method('getSite')->will($this->returnValue($mockSite));
+        $mockDomainRepository->expects($this->atLeastOnce())->method('findOneByActiveRequest')->will($this->returnValue($mockDomain));
+
+        $mockSiteRepository = $this->getMockBuilder(SiteRepository::class)->disableOriginalConstructor()->getMock();
+        $mockSiteRepository->expects($this->any())->method('findFirstOnline')->will($this->returnValue(null));
+
+        $contentContextFactory = $this->getMock(ContentContextFactory::class, [
```
Comment (kdambekalns) [inline TYPO3.Neos/Tests/Unit/Domain/Service/ContentContextFactoryTest.php]: `getMockBuilder(ContentContextFactory::class)->setConstructorArgs([...])->getMock()`

File: TYPO3.Neos/Tests/Unit/Service/PublishingServiceTest.php [hunk 3]
```diff
@@ -104,12 +104,12 @@ public function setUp()
         $this->inject($this->publishingService, 'contextFactory', $this->mockContextFactory);
 
         $this->mockDomainRepository = $this->getMockBuilder('TYPO3\Neos\Domain\Repository\DomainRepository')->disableOriginalConstructor()->getMock();
-        $this->inject($this->publishingService, 'domainRepository', $this->mockDomainRepository);
+//        $this->inject($this->publishingService, 'domainRepository', $this->mockDomainRepository);
```
Comment (kdambekalns) [inline TYPO3.Neos/Tests/Unit/Service/PublishingServiceTest.php]: ?

File: TYPO3.Neos/Tests/Unit/Service/PublishingServiceTest.php [hunk 4]
```diff
@@ -104,12 +104,12 @@ public function setUp()
...
 
         $this->mockSiteRepository = $this->getMockBuilder('TYPO3\Neos\Domain\Repository\SiteRepository')->disableOriginalConstructor()->getMock();
         $this->mockSite = $this->getMockBuilder('TYPO3\Neos\Domain\Model\Site')->disableOriginalConstructor()->getMock();
         $this->mockSiteRepository->expects($this->any())->method('findFirstOnline')->will($this->returnValue($this->mockSite));
-        $this->inject($this->publishingService, 'siteRepository', $this->mockSiteRepository);
+//        $this->inject($this->publishingService, 'siteRepository', $this->mockSiteRepository);
```
Comment (kdambekalns) [inline TYPO3.Neos/Tests/Unit/Service/PublishingServiceTest.php]: ?

**Comment by kitsunet** [comment]
Continued in #578

**Pull request closed without merging**